packages/grafana-sql/src/components/NumberInput.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+import { Input } from '@grafana/ui/src/components/Input/Input';
+
+type NumberInputProps = {
+  value: number;
+  defaultValue: number;
+  onChange: (value: number) => void;
+  width: number;
+  placeholder?: string;
+  disabled?: boolean;
+};
+
+export function NumberInput({ value, defaultValue, onChange, width, placeholder, disabled }: NumberInputProps) {
+  const [isEmpty, setIsEmpty] = React.useState(false);
+  return (
+    <Input
+      type="number"
```

Comment:
Is `defaultValue` not required as an attribute of `Input`?

Reply:
React actually complains when using both props together (controlled vs uncontrolled)